Update documentation to reflect k-NN FAISS AVX512 support (#8307)
opensearch-project/documentation-website: _search-plugins/knn/knn-index.md

### x64 architecture
<!-- vale on -->

> For the x64 architecture, three different versions of the Faiss library are built and shipped with the artifact:

Reviewer: for the version, let us mention it as 2.18.0

> For the x64 architecture, three different versions of the Faiss library are built and shipped with the artifact:

Reviewer: Shall we add a note showing the order of performance avx512>avx2>generic ?

Author: I added a note specifying order of performance below the list of libraries (line 69). Let me know if you think it would be better directly after line 63 instead.

Reviewer: It needs to be added as a note. @kolchfa-aws can you help to fix it when you are reviewing it later

Reviewer 2: Sure, will do.

- `libopensearchknn_faiss.so`: The non-optimized Faiss library without SIMD instructions. 
- `libopensearchknn_faiss_avx2.so`: The Faiss library that contains AVX2 SIMD instructions.

> - `libopensearchknn_faiss_avx512.so`: The Faiss library that contains AVX512 SIMD instructions. 

Reviewer: Can you move avx512.so above avx2.so

> The Faiss library order of performance is AVX512, AVX2, generic.

Reviewer 2: ```suggestion
When using the Faiss library, the performance ranking is: AVX512 > AVX2 > no optimization.
{: .note }

```

> If your hardware supports AVX2, the k-NN plugin loads the `libopensearchknn_faiss_avx2.so` library at runtime.

Reviewer: ```suggestion
If your hardware supports AVX2 but doesn't support AVX512, the k-NN plugin loads the `libopensearchknn_faiss_avx2.so` library at runtime.
```

> If your hardware supports AVX512, the k-NN plugin loads the `libopensearchknn_faiss_avx512.so` library at runtime.

Reviewer 2: ```suggestion
If your hardware supports AVX512, the k-NN plugin loads the `libopensearchknn_faiss_avx512.so` library at runtime.

```

> If your hardware supports AVX512, the k-NN plugin loads the `libopensearchknn_faiss_avx512.so` library at runtime.

Reviewer: Can you move this line above line 69 because the order of preference is AVX512 > AVX2> Generic

> To disable AVX2 and AVX512 and load the non-optimized Faiss library (`libopensearchknn_faiss.so`), specify the `knn.faiss.avx2.disabled` and `knn.faiss.avx512.disabled` static settings as `true` in `opensearch.yml` (defaults are `false`). 

Reviewer: ```suggestion
To disable AVX512 and AVX2 and load the non-optimized Faiss library (`libopensearchknn_faiss.so`), specify the `knn.faiss.avx2.disabled` and `knn.faiss.avx512.disabled` static settings as `true` in `opensearch.yml` (defaults are `false`). 
```

> To disable AVX512 and AVX2 and load the non-optimized Faiss library (`libopensearchknn_faiss.so`), specify the `knn.faiss.avx512.disabled` and `knn.faiss.avx2.disabled` static settings as `true` in `opensearch.yml` (defaults are `false`).

Reviewer 2: ```suggestion
To disable the AVX512 and AVX2 SIMD instructions and load the non-optimized Faiss library (`libopensearchknn_faiss.so`), specify the `knn.faiss.avx512.disabled` and `knn.faiss.avx2.disabled` static settings as `true` in `opensearch.yml` (by default, both of these are `false`).
```

### x64 architecture
<!-- vale on -->

> For the x64 architecture, the following versions of the Faiss library are built and shipped with the artifact:

Reviewer 3: ```suggestion
For x64 architecture, the following versions of the Faiss library are built and shipped with the artifact:
```

- `libopensearchknn_faiss.so`: The non-optimized Faiss library without SIMD instructions. 

> - `libopensearchknn_faiss_avx512.so`: The Faiss library that contains AVX512 SIMD instructions. 

Reviewer 3: ```suggestion
- `libopensearchknn_faiss_avx512.so`: The Faiss library containing AVX512 SIMD instructions. 
```

> - `libopensearchknn_faiss_avx2.so`: The Faiss library that contains AVX2 SIMD instructions.

Reviewer 3: ```suggestion
- `libopensearchknn_faiss_avx2.so`: The Faiss library containing AVX2 SIMD instructions.
```

> When using the Faiss library, the performance ranking is: AVX512 > AVX2 > no optimization.

Reviewer 3: ```suggestion
When using the Faiss library, the performance ranking is as follows: AVX512 > AVX2 > no optimization.
```
